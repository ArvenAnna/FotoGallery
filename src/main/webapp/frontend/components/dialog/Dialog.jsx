import React from 'react';
import {CrossIcon} from "../Icons";
import './dialog.less';


class Dialog extends React.Component {

    constructor(props) {
        super(props);
    }


    render() {
        return <div className="Dialog">
                <CrossIcon className="cross" onClick={this.props.onClose}/>
                <div className="Dialog_text">{this.props.text}</div>
                <button className="Dialog_ok_button" onClick={this.props.onClick}>ok</button>
            </div>
    }
}

export default Dialog;
