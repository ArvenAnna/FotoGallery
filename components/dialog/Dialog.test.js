import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

// jest.mock("react-redux/es/connect/connect", () => ({
// }));

import Dialog from './Dialog';

describe('Dialog', () => {

    it('should render correctly', () => {
        const output = shallow(
            <Dialog onClick={() => {}} onClose={() => {}} text="mock text"/>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});