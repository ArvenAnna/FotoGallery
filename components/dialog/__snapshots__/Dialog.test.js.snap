// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog should render correctly 1`] = `
<div
  className="Dialog"
>
  <styled.
    className="cross"
    onClick={[Function]}
  />
  <div
    className="Dialog_text"
  >
    mock text
  </div>
  <button
    className="Dialog_ok_button"
    onClick={[Function]}
  >
    ok
  </button>
</div>
`;
